forge-ai/src/main/java/forge/ai/PlayerControllerAi.java

```diff
@@ -105,8 +102,99 @@ public boolean isAI() {
 
     @Override
     public List<PaperCard> sideboard(Deck deck, GameType gameType, String message) {
-        // AI does not know how to sideboard
-        return null;
+        if (!getAi().getBooleanProperty(AiProps.SIDEBOARDING_ENABLE)) {
+            return null;
+        }
+
+        boolean sbLimitedFormats = getAi().getBooleanProperty(AiProps.SIDEBOARDING_IN_LIMITED_FORMATS);
+        boolean sbSharedTypesOnly = getAi().getBooleanProperty(AiProps.SIDEBOARDING_SHARED_TYPE_ONLY);
+        boolean sbPlaneswalkerException = getAi().getBooleanProperty(AiProps.SIDEBOARDING_PLANESWALKER_EQ_CREATURE);
+        int sbChanceOnWin = getAi().getIntProperty(AiProps.SIDEBOARDING_CHANCE_ON_WIN);
+        int sbChancePerCard = getAi().getIntProperty(AiProps.SIDEBOARDING_CHANCE_PER_CARD);
+
+        if (!sbLimitedFormats && gameType.isCardPoolLimited()) {
+            return null;
+        }
+
+        if (MyRandom.getRandom().nextInt(100) < sbChanceOnWin) {
+            GameOutcome lastOutcome = brains.getGame().getMatch().getOutcomeById(brains.getGame().getId() - 1);
```

Comment:
I think this calculation would risk a NPE, since the global ID might get increased by more from other matches 💣 
better grab the `lastOutcome` field directly?

Reply:
Agreed, done :) Also, fixed the related logic while I was at it.